File: 04/exemplo.py
# 1. Tipagem dinâmica: a variável 'dado' primeiro armazena um inteiro.
dado = 10
print(f"O valor é {dado} e o tipo é {type(dado)}")

# 2. Agora, a mesma variável armazena um texto. O tipo muda dinamicamente.
dado = "dez"
print(f"O valor agora é '{dado}' e o tipo mudou para {type(dado)}")

# 3. Tipagem forte em ação: Python não permite operações ambíguas.
# A linha abaixo, se fosse descomentada, causaria um ERRO DE EXECUÇÃO (TypeError),
# pois Python não sabe se deve somar ou concatenar.
# resultado = 10 + "5"
# print(resultado)

print("\nPython é fortemente tipado: uma operação como 10 + '5' resulta em erro.")

File: 04/exemplo.js
// 1. Tipagem dinâmica: 'valor' é um número.
let valor = 10;
console.log(`O valor é ${valor} e o tipo é ${typeof valor}`);

// 2. Agora a mesma variável guarda um texto.
valor = "dez";
console.log(`O valor agora é '${valor}' e o tipo mudou para ${typeof valor}`);

// 3. Tipagem fraca em ação (coerção de tipo).
// O JavaScript converte o número 10 para string e concatena,
// em vez de lançar um erro.
let somaInesperada = 10 + "5";

console.log(`\nJavaScript é fracamente tipado:`);
console.log(`O resultado de 10 + "5" é: ${somaInesperada}`);
console.log(`O tipo do resultado é: ${typeof somaInesperada}`);